lib/CMakeLists.txt

```diff
@@ -13,487 +12,91 @@
-    render/pxrUsdMayaGL/sceneDelegate.h
-    render/pxrUsdMayaGL/shapeAdapter.h
-    render/pxrUsdMayaGL/softSelectHelper.h
-    render/pxrUsdMayaGL/usdProxyShapeAdapter.h
-    render/pxrUsdMayaGL/userData.h
-)
-
-list(APPEND mayaUsdPxVP20_headers
-    render/px_vp20/glslProgram.h
-    render/px_vp20/utils.h
-    render/px_vp20/utils_legacy.h
-)
-
-add_library(${LIBRARY_NAME} SHARED
-    ${mayaUsd_src}
-)
-
-set(compile_definitions_list
-    MAYAUSD_MACROS_EXPORT
-    MAYAUSD_CORE_EXPORT
-    MFB_PACKAGE_NAME="${LIBRARY_NAME}"
-    MFB_ALT_PACKAGE_NAME="${LIBRARY_NAME}"
-    MFB_PACKAGE_MODULE=mayaUsd
-)
-
-if(UFE_FOUND)
-    list(APPEND compile_definitions_list WANT_UFE_BUILD=1)
-endif()
-if(IS_MACOSX)
-    list(APPEND compile_definitions_list OSMac_)
-endif()
-
-target_compile_definitions(${LIBRARY_NAME} PRIVATE
-    ${compile_definitions_list}
+# -----------------------------------------------------------------------------
+# compile definitions
+# -----------------------------------------------------------------------------
+target_compile_definitions(${PROJECT_NAME} 
+    PRIVATE
+        MAYAUSD_MACROS_EXPORT
+        MAYAUSD_CORE_EXPORT
+        MFB_PACKAGE_NAME="${PROJECT_NAME}"
+        MFB_ALT_PACKAGE_NAME="${PROJECT_NAME}"
+        MFB_PACKAGE_MODULE=mayaUsd
+        $<$<BOOL:${UFE_FOUND}>:WANT_UFE_BUILD>
+        $<$<BOOL:${IS_MACOSX}>:OSMac_>
 )
 
 # Some of the UFE classes are exporting STL classes which causes this warning.
 if(UFE_FOUND AND MSVC)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4251")
     string(REPLACE " " ";" REPLACED_FLAGS ${CMAKE_CXX_FLAGS})
-    target_compile_options(${LIBRARY_NAME} INTERFACE ${REPLACED_FLAGS})
-endif()
-
-set(include_directories_list
-    ${MAYA_INCLUDE_DIRS}
-    ${PXR_INCLUDE_DIRS}
-    ${CMAKE_BINARY_DIR}/include
-)
-
-if(UFE_FOUND)
-    list(APPEND include_directories_list ${UFE_INCLUDE_DIR})
-endif()
-
-target_include_directories( ${LIBRARY_NAME} PUBLIC
-    ${include_directories_list}
-)
-
-target_link_libraries(${LIBRARY_NAME}
-    ${Boost_FILESYSTEM_LIBRARY}
-    ${Boost_SYSTEM_LIBRARY}
-    ${MAYA_Foundation_LIBRARY}
-    ${MAYA_OpenMayaAnim_LIBRARY}
-    ${MAYA_OpenMaya_LIBRARY}
-    ${MAYA_OpenMayaRender_LIBRARY}
-    ${MAYA_OpenMayaUI_LIBRARY}
-    ar
-    gf
-    hd
-    hdx
-    js
-    kind
-    plug
-    sdf
-    tf
-    trace
-    usd
-    usdGeom
-    usdImaging
-    usdImagingGL
-    usdLux
-    usdRi
-    usdShade
-    usdSkel
-    usdUtils
-    vt
+    target_compile_options(${PROJECT_NAME} INTERFACE ${REPLACED_FLAGS})
+endif()
+
+# -----------------------------------------------------------------------------
+# include directories
+# -----------------------------------------------------------------------------
+target_include_directories(${PROJECT_NAME} 
+    PUBLIC
+        ${CMAKE_BINARY_DIR}/include
+        ${MAYA_INCLUDE_DIRS}
+        ${PXR_INCLUDE_DIRS}
+        $<$<BOOL:${UFE_FOUND}>:${UFE_INCLUDE_DIR}>
+    PRIVATE
+        ${CMAKE_CURRENT_SOURCE_DIR}
+)
```

Comment:
Not sure if this is slated for a later phase, but it would be great to verify that all of the named targets we're using with `target_link_libraries()` properly identify their interface include directories so that we don't need to specify them here. I **think** the pxr libraries should be doing that already, and maybe we should be setting up `Maya::*` and `UFE::*` targets in the Find modules.